### DISCUSSION TITLE
why the self time of rokect mq consumer is so long and no detail in trace panel?

### DISCUSSION THREAD

**Discussion by bigwalnut** [apache/skywalking#8169] [Q&A]
![image](https://user-images.githubusercontent.com/42861474/142798617-03562bd3-3543-4489-a7a6-3d48241c00f9.png)

As the above pic shown, "RocketMQ/earchms-apiMessageListenerTopic-dev/Consumer" is a rocket mq comsumer, the execution time of it is 31521ms, and the self time of it is 28024ms, but the sum of all other step's self time is less than 4000ms, why the self time of rokect mq consumer is so long，what happend? I am not familiar with rocket mq.

**Comment by wu-sheng** [comment]
Why it must have? Tracing in default only covers RPCs relative time cost, you have many other logic processed when consume a message. Such as large loop, lock, wait for another thread. 
Profiling is provided to determine this. 

> why the self time of rokect mq consumer is so long，what happend? I am not familiar with rocket mq.

BTW, this question doesn't make any sense to SkyWalking. We don't have relationship with them.

**Reply by bigwalnut** [reply]
Thank you for your answer！I asked this question because it generated an alarm, and our developer said that he didn’t know why the mq consumer deal with it for so long. I want to know how skywalking agent to report the trace "RocketMQ/earchms-apiMessageListenerTopic-dev/Consumer"， Where to monitor？from where to where in our system's code.

**Reply by wu-sheng** [reply]
I think this should not be answered by us. The source codes have been opened, where we collect duration is open to you. 
This kind of slow execution should be located by developers rather than ops team. That is the point of tracing.

**Reply by bigwalnut** [reply]
Ok, thanks for your advice. We try to solve it ourselves.
